lib/puppet/http/service/file_server.rb

```diff
@@ -0,0 +1,103 @@
+require 'puppet/file_serving/metadata'
+
+class Puppet::HTTP::Service::FileServer < Puppet::HTTP::Service
+  API = '/puppet/v3'.freeze
+
+  EXCLUDED_FORMATS = [:yaml, :b64_zlib_yamli, :dot]
+
+  def initialize(client, server, port)
+    url = build_url(API, server || Puppet[:server], port || Puppet[:masterport])
+    super(client, url)
+  end
+
+  def get_file_metadata(mount_point:, path:, environment:,
+                        links: :manage, checksum_type: Puppet[:digest_algorithm], source_permissions: :ignore,
+                        ssl_context: nil)
+
+    model = Puppet::Transaction::Report
+    network_formats = model.supported_formats - EXCLUDED_FORMATS
+    mime_types = network_formats.map { |f| model.get_format(f).mime }
```

Comment:
Since the list of supported formats is static, recommend moving this to a private method and memoizing the result. Then `get_file_metadatas` (plural) below can call that method too.